whole-repo-tests/test_rst_is_valid.py

```diff
@@ -41,3 +42,10 @@ def test_passes_rst_lint():
 
 def test_passes_flake8():
     pip_tool('flake8', '--select=W191,W291,W292,W293,W391', *ALL_RST)
+
+
+def test_crossref_type():
+    """Check that there are no examples of a common typo in the docs."""
+    # Note: not just docs directory, but also RELEASE.rst and docstrings.
+    # Exits with 0 if matches found, so assert non-zero exit code.
+    assert subprocess.call(['git', 'grep', '-nP', "'`~?hypotheses'"])
```

Comment:
I prefer automated commands to use their full flag names, for clarity, i.e.:

```python
subprocess.call(['git', 'grep', '--line-number', '--perl-regexp', "'`~?hypothesis'"])
```

Also, a quick explanation of what the regex is doing would be helpful. (And do you need the single quote marks?)